io/include/traccc/io/read_geometry.hpp

```diff
@@ -27,4 +27,19 @@ namespace traccc::io {
 geometry read_geometry(std::string_view filename,
                        data_format format = data_format::csv);
 
+/// Read in the detector geometry description from a detector object
+template <typename detector_t>
+geometry alt_read_geometry(const detector_t& det) {
+
+    const auto transforms = det.transform_store();
+    const auto surfaces = det.surfaces();
+    std::map<traccc::geometry_id, traccc::transform3> maps;
```

Comment:
Don't remember. Maybe we need it for binary search later

Reply:
Currently I think I am only using it in `read_spacepoints.cpp` for single lookups. So yeah I think it could be unordered map for my use case but it might be used differently in other places

Reply:
OK